users/views.py

```diff
@@ -1,3 +1,59 @@
-from django.shortcuts import render
+import json, re, bcrypt
 
-# Create your views here.
+from django.http  import JsonResponse
+from django.views import View
+
+from .models import User
+
+
+class SignUpView(View):
+    def post(self, request):
+        try:
+            data           = json.loads(request.body)
+            name           = data['name']
+            email          = data['email']
+            password       = data['password']
+            mobile_number  = data['mobile_number']
+            birthday       = data.get('birthday')
+
+            REGEX_PASSWORD = '^(?=.*[A-Za-z])(?=.*\d)(?=.*[?!@#$%*&])[A-Za-z\d?!@#$%*&]{8,}$'
+            REGEX_EMAIL    = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+            REGEX_BIRTHDAY = '^(19[0-9][0-9]|20\d{2})-(0[0-9]|1[0-2])-(0[1-9]|[1-2][0-9]|3[0-1])$'
+            REGEX_MOBILE   = '^\d{3}-\d{3,4}-\d{4}$'
+
+            checkRegex(REGEX_EMAIL, email)
+
+            if User.objects.filter(email=email).exists():
+                return JsonResponse({'MESSAGE': 'ALREADY_EXISTS_EMAIL'}, status = 400)
+            
+            checkRegex(REGEX_PASSWORD, password)
+                
+            if User.objects.filter(mobile_number=mobile_number).exists():
+                return JsonResponse({'MESSAGE': 'ALREADY_EXISTS_MOBILE_NUMBER'}, status = 400)
+
+            checkRegex(REGEX_MOBILE, mobile_number)
+
+            if birthday: 
+                if not re.compile(REGEX_BIRTHDAY).match(birthday):
+                    raise ValueError
+
+            hashed_password = bcrypt.hashpw(password.encode("utf-8"), bcrypt.gensalt()).decode("utf-8")
+
+            User.objects.create(
+                name          = name,
+                email         = email,
+                password      = hashed_password,
+                mobile_number = mobile_number,
+                birthday      = birthday
+            )
+            return JsonResponse({'message': 'sucess'}, status=201)
+
+        except KeyError as e:
+            return JsonResponse({'message': f"(KEY_ERROR:{e})"}, status=400)
+        
+        except ValueError:
+            return JsonResponse({"message": 'Format is not valid'}, status=400)
+            
+def checkRegex(REGEX, value):
+    if not re.compile(REGEX).match(value):
```

Comment:
훌륭합니다 정훈님 👍

다만 다음과 같은 정규식을 통한 validation check 같은 경우는 현재 구현사항에 포함되지 않더라도 추후에 `users` app뿐만이 아닌 다른 곳에서도 충분히 사용 될 수 있기 때문에(ex. `orders` app에서 비회원이 주문서를 작성하는 경우와 같이) `core/utils.py`에 작성하여 모듈로 import 하여 사용하는 것이 좋습니다.

추가로 정규표현식을 사용할 경우, 자주 사용되는 경우가 아니라면 `compile()`해서 사용하지 않고, `match()`를 사용하는 것을 공식 문서에서 추천하고 있습니다.

```suggestion
    if not re.match(REGEX, value):
```

Reply:
메소드에서 로직을 사용하는것과 바로 사용하는것에 차이가 있을까요?